src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,128 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_SIZE = 10;
+    private int size;
+    private T[] listArray;
+
+    public ArrayList() {
+        size = 0;
+        listArray = (T[]) new Object[DEFAULT_ARRAY_SIZE];
+    }
+
     @Override
     public void add(T value) {
+        if (size >= listArray.length) {
+            listArray = extendArray();
+        }
 
+        listArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size + 1 || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
+
+        if (size + 1 >= listArray.length) {
+            listArray = extendArray();
+        }
 
+        size++;
+
+        T[] newArray = (T[]) new Object[listArray.length];
+        for (int i = 0; i <= size; i++) {
+            if (i < index) {
+                newArray[i] = listArray[i];
+            }
+
+            if (i > index) {
+                newArray[i] = listArray[i - 1];
+            }
+        }
+
+        newArray[index] = value;
+        listArray = newArray;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            if (list.get(i) != null) {
```

Comment:
Checking for null isn't necessary, as ArrayList allows null elements. If the list contains null elements, they should be added to the ArrayList.